src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -597,19 +601,17 @@ impl Net {
                 .add(tx_queue.len(mem).into());
             let head_index = head.index;
             // Parse IoVecBuffer from descriptor head
-            let buffer = match IoVecBuffer::from_descriptor_chain(head) {
-                Ok(buffer) => buffer,
-                Err(_) => {
-                    self.metrics.tx_fails.inc();
-                    tx_queue
-                        .add_used(mem, head_index, 0)
-                        .map_err(DeviceError::QueueError)?;
-                    continue;
-                }
-            };
+            // SAFETY: This descriptor chain is only loaded into this buffer
```

Comment:
Let's elaborate on these - virtio requests are handled sequentially, so no two iovecbuffers are ever "live" at the same time, meaning this one really has exclusive ownership over the memory (well, from rust's side.. the guest can do as it pleases)

Reply:
We should `clear()` this buffer as soon as we're done using it (e.g. at the end of this function)

Reply:
Sounds great thanks I'll update the comments and clear at the end thanks